### PR TITLE
Support Triton fp8 e5m2 kv cache

### PR DESCRIPTION
**Pull request by ispobock**
## Motivation

Previously fp8 kv cache for Flashinfer was supported in https://github.com/sgl-project/sglang/pull/1204.
This PR support this feature for Triton runtime.

## Evaluation
DeepSeek-Coder-V2-Lite-Instruct
| backend    | kv cache dtype | gsm8k flexible-extract | gsm8k strict-match |
| ---------- | -------------- | ---------------------- | ------------------ |
| Flashinfer | bf16           | 0.7703                 | 0.7627             |
| Flashinfer | fp8_e5m2       | 0.7635                 | 0.7491             |
| Triton     | fp16           | 0.7839                 | 0.7695             |
| Triton   | fp8_e5m2       | 0.7627                 | 0.7498             |

Meta-Llama-3.1-8B-Instruct
| backend    | kv cache dtype | gsm8k flexible-extract | gsm8k strict-match |
| ---------- | -------------- | ---------------------- | ------------------ |
| Flashinfer | bf16           | 0.7771                 | 0.7127             |
| Flashinfer | fp8_e5m2       | 0.7650                 | 0.7043             |
| Triton     | fp16           | 0.7817                 | 0.7202             |
| Triton    | fp8_e5m2       | 0.7710                 |   0.7498         |

```
python3 -m sglang.launch_server --model-path deepseek-ai/DeepSeek-Coder-V2-Lite-Instruct --port 30000 --trust-remote-code --host 0.0.0.0 --tp=1 --enable-mla --kv-cache-dtype fp8_e5m2

lm_eval --model local-completions --tasks gsm8k --model_args model=deepseek-ai/DeepSeek-Coder-V2-Lite-Instruct,base_url=http://0.0.0.0:30000/v1/completions,num_concurrent=128,max_retries=3,tokenized_requests=False
```

## Checklist

- [ ] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/en/contributor_guide.md).
- [ ] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/en/contributor_guide.md).
- [ ] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

**Comment by zhyncs** [comment]
Nice work!

**Comment by zhyncs** [comment]
ref https://github.com/sgl-project/sglang/issues/1156

**Review by zhyncs — APPROVED** [review]

**Comment by zhyncs** [comment]
hold on, we should also verify on H100

File: python/sglang/srt/layers/extend_attention.py [hunk 1]
```diff
@@ -276,9 +276,17 @@ def extend_attention_fwd(
     BLOCK_DV = Lv
 
     if CUDA_CAPABILITY[0] >= 9:
-        BLOCK_M, BLOCK_N = (128, 64)
+        if Lq <= 256:
+            BLOCK_M, BLOCK_N = (128, 64)
+        else:
+            BLOCK_M, BLOCK_N = (32, 64)
```
Comment (zhyncs) [inline python/sglang/srt/layers/extend_attention.py:282]: note: 32,64 is better than 128,64 for DeepSeek V2

**Comment by yzh119** [comment]
Note: the current fp8 prefill kernels in flashinfer is a mixed-precision kernel (upcast fp8 K/V to fp16 inside kernels and use fp16 tensor cores), which is slow because of the online dequantization overhead, and it's different from the triton kernel logic here (where you downcast fp16 query to fp8 and use fp8 tensor cores).

The native fp8 prefill attention kernel in flashinfer will land together with fa3, where we use fp8 tensor cores, but the accuracy landscape might change (and might align with your triton implementation).

**Comment by zhyncs** [comment]
> Note: the current fp8 prefill kernels in flashinfer is a mixed-precision kernel (upcast fp8 K/V to fp16 inside kernels and use fp16 tensor cores), which is slow because of the online dequantization overhead, and it's different from the triton kernel logic here (where you downcast fp16 query to fp8 and use fp8 tensor cores).
> 
> The native fp8 prefill attention kernel in flashinfer will land together with fa3, where we use fp8 tensor cores, but the accuracy landscape might change (and might align with your triton implementation).

Looking forward to it very much!

**Pull request merged**